src/sortStudents.ts

```diff
@@ -1,16 +1,75 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname:string;
+  age: number;
+  married: boolean;
+  grades:number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc'|'desc';
 
+function avg(grades:number[]):number {
+  return grades.reduce((sum:number, grade:number) => sum + grade, 0)
+  / grades.length;
+}
+
+export function sortStudents(
+
+  students :Student[],
+  sortBy : SortType,
+  order:SortOrder,
+)
+  :Student[] {
+  const sortedArray:Student[] = [...students].map((student: Student) => {
+    return { ...student };
+  });
```

Comment:
map returns new array, first copying is redundant here
```suggestion
  const sortedArray:Student[] = students.map((student: Student) => {
    return { ...student };
  });
```